File: src/main/java/fr/adrienbrault/idea/symfony2plugin/assetMapper/dict/MappingFileEnum.java
package fr.adrienbrault.idea.symfony2plugin.assetMapper.dict;

import org.jetbrains.annotations.NotNull;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public enum MappingFileEnum {
    IMPORTMAP, INSTALLED;

    public static MappingFileEnum fromString(@NotNull String text) {
        if (text.equalsIgnoreCase("importmap.php")) {
            return IMPORTMAP;
        }

        if (text.equalsIgnoreCase("installed.php")) {
            return INSTALLED;
        }

        throw new RuntimeException("invalid filename");
    }
}

File: src/main/kotlin/fr/adrienbrault/idea/symfony2plugin/assetMapper/dict/MappingFileEnum.kt
package fr.adrienbrault.idea.symfony2plugin.assetMapper.dict

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
enum class MappingFileEnum {
    IMPORTMAP, INSTALLED;

    companion object {
        @JvmStatic
        fun fromString(text: String): MappingFileEnum {
            if (text.equals("importmap.php", ignoreCase = true)) {
                return IMPORTMAP
            }

            if (text.equals("installed.php", ignoreCase = true)) {
                return INSTALLED
            }

            throw RuntimeException("invalid filename")
        }
    }
}
